daemon/cluster/executor/container/container.go

```diff
@@ -72,13 +72,8 @@ func (c *containerConfig) spec() *api.ContainerSpec {
 }
 
 func (c *containerConfig) name() string {
-	if c.task.Annotations.Name != "" {
```

Comment:
Don't remove this. We want to be able to name tasks this way in the future. Removing this makes backwards compatibility impossible.